AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt

```diff
@@ -2738,8 +2741,28 @@ open class DeckPicker :
             updateDeckList()
         }
     }
-}
 
+    // Scoped Storage migration
+    fun postponeMigration() {
+        getSharedPrefs(baseContext)
+            .edit { putLong(POSTPONE_MIGRATION_KEY, TimeManager.time.intTime()) }
+    }
+
+    /**
+     * Whether we can offer the migration at current time.
+     * That is, we have not postponed in the last 10 days.
+     */
+    fun shouldWaitMoreToOfferToMigrate(): Boolean {
+        val secondWhenMigrationWasPostponedLast = getSharedPrefs(baseContext).getLong(POSTPONE_MIGRATION_KEY, -1)
+        if (secondWhenMigrationWasPostponedLast == -1L) return false
+        val delaySyncPostpone = TimeManager.time.intTime() - secondWhenMigrationWasPostponedLast
+        if (delaySyncPostpone > 10 * 24 * 60 * 60) {
```

Comment:
More generic comment, really just so that as we discuss timing we only have to change one spot.
Use constant

```suggestion
     * That is, the last time the user postponed was longer than our postpone time.
     */
    fun shouldWaitMoreToOfferToMigrate(): Boolean {
        val secondWhenMigrationWasPostponedLast = getSharedPrefs(baseContext).getLong(POSTPONE_MIGRATION_KEY, -1)
        if (secondWhenMigrationWasPostponedLast == -1L) return false
        val delaySyncPostpone = TimeManager.time.intTime() - secondWhenMigrationWasPostponedLast
        if (delaySyncPostpone > POSTPONE_MIGRATION_INTERVAL_DAYS * 24 * 60 * 60) {
```